6_nodejs/week3/app.js

```diff
@@ -0,0 +1,57 @@
+import knex from 'knex'
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "my-secret-pw",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import 'dotenv/config'
+import express from 'express'
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort;
+    const allowedColumnsForSort = [
+        "first_name",
+        "last_name",
+        "email",
+        "phone",
+        "id"
+      ];
+      if (orderBy.length > 0 && allowedColumnsForSort.includes(orderBy)) {
+        query = query.orderByRaw(orderBy); 
+      }
```

Comment:
Your solution works just fine for letting the user decide which column to sort the result on. However, in the original code, the user also had the option to determine whether the sorting should be ascending or descending using the second part of the original parameter.

Currenlty your new solution removed this functionality. We prefer to never sacrifice functionality. Thankfully, you can easily add this functionality back to your code by having another query parameter, let's say `orderDir`. Remember though, new parameter = new SQL injection possibility! Make sure to allowlist that parameter too. 😉 🛑